primitives/src/runtime_api.rs

```diff
@@ -0,0 +1,20 @@
+use crate::constants::transaction::MaxExtrinsicSize;
+use codec::Codec;
+use frame_support::BoundedVec;
+use sp_weights::Weight;
+
+sp_api::decl_runtime_apis! {
+	pub trait FeeEstimationApi<AccountId, AssetId, Balance>
```

Comment:
Naming. 

I would go with slightly different naming .

TransactionFeeAPI

and methods:
- estimate_fee
- estimate_uxt_fee

I let @jak-pan share his thoughts here on naming,